demos/LowAllocationWebServer/ByteSpanExtensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+
+namespace LowAllocationServer
+{
+    public static class ByteSpanExtensions
+    {
+        internal static ByteSpan SliceTo(this ByteSpan buffer, char terminator, out int consumedBytes)
+        {
+            return buffer.SliceTo((byte) terminator, out consumedBytes);
```

Comment:
Does not change the fact that the implementation might will not work for some characters. We should make it work by checking if the char cannot be represented by one byte.